tsconfig.json

```diff
@@ -0,0 +1,23 @@
+{
+  "compilerOptions": {
+    "target": "esnext",
+    "module": "commonjs",
+    "resolveJsonModule": true,
+    "allowJs": true,
+    "checkJs": false,
+    "outDir": "./dist",
+    "esModuleInterop": true,
+    "forceConsistentCasingInFileNames": true,
+    "strict": false,
+    "skipLibCheck": true
```

Comment:
`checkJs`, `strict` and lib check sound like things we'd prefer to be enabled. Is there a good reason to disable them? Is it maybe just temporary because it's not yet a complete rewrite?

Reply:
allowJs and checkJs can now be removed. Since we don't have any js code anymore. This would be a bridge between having both `js` and `ts` files in the same project. All files are now `.ts`  so this can be removed. I will remove them.

Reply:
strict can now be turned on, i agree.

Reply:
Oh, just found the comments in b0ed53d1f3cbf8978260c12cdd9af40ba38e6c1b. I think it would have been better to put these as comments in the code. And maybe with a `// TODO` or `// FIXME` if that's not meant to be a permanent thing.

Reply:
@cameel using strict: true will result in 40-170 errors, I think having this off for the first stage would be best.

Errors are mostly `Parameter 'error' implicitly has an 'any' type.` but allowing any types results in 30 invalid errors that would have to be resolved first.

Reply:
ok, it's reasonable to disable it for now then. Just add a `FIXME`. And maybe an issue - but the only if the PR gets merged and we decide not to do it right away for some reason.

Reply:
checkJs and allowJs have now been removed since the project is entirely `.ts` files and a comment has been added regarding `strict`.